nexus/tests/integration_tests/oximeter.rs

```diff
@@ -146,12 +146,52 @@ async fn test_oximeter_reregistration() {
     //
     // At this point, Oximeter will still be collecting, so we poll for data and verify that
     // there's no new data in ClickHouse.
+    //
+    // Specifically, we grab a timestamp before dropping the producer, and assert that any data in
+    // the timeseries is from before that timestamp.
+    let time_producer_dropped = chrono::Utc::now();
     drop(context.producer);
     let new_timeseries =
         retrieve_timeseries().await.expect("Failed to retrieve timeseries");
     assert_eq!(new_timeseries.len(), 1);
     let new_timeseries = new_timeseries.into_iter().next().unwrap();
-    assert_eq!(timeseries, new_timeseries);
+
+    // The test here is a bit complicated, so deserves some explanation.
+    //
+    // We'd like to just assert that the old and new timeseries are equal. That's not quite right,
+    // since data may be somewhere on its way to ClickHouse before we drop the producer, e.g. in
+    // the collector's memory. So we first check that the original timeseries is a prefix of the
+    // new timeseries. Then we check that any remaining measurements are no later than the
+    // timestamp immediately prior to dropping the producer.
+    fn check_following_timeseries(
```

Comment:
Might you want #[track_caller] on this function so that assertion failures would indicate which of the locations was problematic? or would it be better to just enable stacks at that point